### ISSUE TITLE
A buggy implementation of `randcat` for numpy

### ISSUE THREAD

**Issue by stoprightthere** [wesselb/lab#10]
The implementation of `lab.numpy.random.randcat` is incorrect.

**Example**
```python
import lab as B
import numpy as np

ps = np.r_[1., 0., 0., 0.]
B.randcat(ps, 1)    # should be 0 because it has probability 1
```
Expected output: `array([0])`
Got output: `array([3])`

**Suggested fix**: use `np.random.choice`.

**Comment by wesselb** [comment]
Hey @stoprightthere ! Thanks for raising this, and I'm sorry for the slow reply. Been away on holiday and to a conference, and I'm only now catching up on email.

Ouch, this is an extremely embarassing bug. This is _definitely_ not the intended behaviour. I will look into it!

**Comment by wesselb** [comment]
This should be fixed on the latest release `v1.5.4`. I've added a test that explicitly checks your above example.

**Comment by stoprightthere** [comment]
Hey @wesselb , thanks for fixing the issue!
